src/PrestaShopBundle/Form/ErrorMessage/LabelProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\ErrorMessage;
+
+use PrestaShopBundle\Controller\Exception\FieldNotFoundException;
+use Symfony\Component\Form\FormInterface;
+
+class LabelProvider
+{
+    /**
+     * @param FormInterface $form
+     * @param string $fieldName
+     *
+     * @return string
+     * @throws FieldNotFoundException
+     */
+    public function getLabel(FormInterface $form, string $fieldName): string
+    {
+        $view = $form->createView();
+        foreach ($view->children as $child) {
+            if ($fieldName === $child->vars['name']) {
```

Comment:
```suggestion
            if ($fieldName === $child->vars['name']) {
```

Please check `'name'` key :)

Reply:
Could you elaborate? I don't quite get the problem here.

Reply:
My guess is that Progi1984 is worried if `$child->vars` does not contain `name` key.

But maybe such children are object generated by Symfony and the `name` key inside ` $child->vars` array is provided by Symfony by default?

Reply:
```suggestion
            if (isset($child->vars['name']) && $fieldName === $child->vars['name']) {
```